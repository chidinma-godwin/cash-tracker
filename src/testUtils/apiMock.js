import {
  loginEndpoint,
  signupEndpoint,
  transactionEndpoint,
  userEndpoint,
} from 'constants/endpoints';
import { createServer, Model } from 'miragejs';

export default function makeServer(environment = 'test') {
  return createServer({
    environment,
    models: {
      user: Model,
      transaction: Model,
    },
    routes() {
      this.get(userEndpoint, () => ({
        user: {
          id: '1',
          username: 'Authenticated User',
          email: 'user@email.com',
          clientsEmail: ['client1@email.com'],
          pendingInvitations: ['invite@email.com'],
          pendingRequests: ['pending1@email.com', 'pending2@email.com'],
        },
      }));
      this.post(signupEndpoint, (schema, request) => {
        const attrs = JSON.parse(request.requestBody);
        return schema.users.create(attrs);
      });
      this.post(loginEndpoint, (schema, request) => {
        const { email, password } = JSON.parse(request.requestBody);
        return schema.users.findBy({ email, password });
      });
      this.get(`${transactionEndpoint}/:id`, (schema, request) => {
        const { id } = request.params;
        return schema.transactions.where(transaction =>
          transaction.participants.includes(id)
        );
      });
      this.post(transactionEndpoint, (schema, request) => {
        const attrs = JSON.parse(request.requestBody);
        return schema.transactions.create(attrs);
      });
    },
  });
}
