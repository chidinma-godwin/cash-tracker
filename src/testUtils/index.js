import React from 'react';
import { SWRConfig } from 'swr';
import { render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';

// eslint-disable-next-line react/prop-types
const Wrapper = ({ children }) => (
  <SWRConfig value={{ dedupingInterval: 0 }}>{children}</SWRConfig>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

const customRenderHook = hook => renderHook(hook, { wrapper: Wrapper });

export * from '@testing-library/react';
export { customRenderHook as renderHook };
export { customRender as render };
