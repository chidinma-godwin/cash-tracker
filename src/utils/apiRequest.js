import {
  unexpected,
  wrongInput,
  notFound,
  unauthenticated,
} from 'constants/errorMessages';

const apiRequest = async (url, body) => {
  let err = '';
  const response = await fetch(url, {
    method: body ? 'POST' : 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  });
  if (response.status === 400) {
    err = wrongInput;
  } else if (response.status === 500) {
    err = unexpected;
  } else if (response.status === 404) {
    err = notFound;
  } else if (response.status === 401) {
    err = unauthenticated;
  } else {
    err = unexpected;
  }
  return { response, err };
};

export default apiRequest;
