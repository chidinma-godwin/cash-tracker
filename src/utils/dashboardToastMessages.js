export const getNotModifiedErrMsg = ({ action, type, setErrMsg }) => {
  if (action === 'add' && type === 'request') {
    setErrMsg('You have already sent a connection request to this user');
  } else if (action === 'add') {
    setErrMsg('You already have an account with this user');
  } else if (action === 'delete' && type === 'request') {
    setErrMsg('You have already cancelled your connection request');
  } else if (action === 'delete' && type === 'account') {
    setErrMsg('You have already deleted your account with this user');
  } else {
    setErrMsg('You have already declined connection invitation from this user');
  }
};

export const getUpdateSuccessfulMessage = ({ action, type, setSuccessMsg }) => {
  if (action === 'add' && type === 'request') {
    setSuccessMsg('Invitation sent successfully');
  } else if (action === 'add') {
    setSuccessMsg('Transaction account successfully confirmed');
  } else if (action === 'delete' && type === 'request') {
    setSuccessMsg('Transaction account request cancelled');
  } else if (action === 'delete' && type === 'account') {
    setSuccessMsg('Transaction account deleted successfully');
  } else {
    setSuccessMsg('Transaction account invitation declined');
  }
};
export const getUpdateFailedMsg = ({ action, type, setErrMsg }) => {
  if (action === 'add' && type === 'request') {
    setErrMsg('Unable to invite this user. Please try again.');
  } else if (action === 'add') {
    setErrMsg('Unable to confirm this account. Please try again.');
  } else if (action === 'delete' && type === 'request') {
    setErrMsg('Unable to cancel request. Please try again.');
  } else if (action === 'delete' && type === 'account') {
    setErrMsg(
      'Unable to delete your account with this user. Please try again.'
    );
  } else {
    setErrMsg('Unable to decline this invitation. Please try again.');
  }
};
