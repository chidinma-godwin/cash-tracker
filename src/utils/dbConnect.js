import mongoose from 'mongoose';

const connection = {};

async function dbConnect() {
  if (connection.isConnected) {
    return;
  }

  /* Connecting to our database */
  const db = await mongoose.connect(process.env.MONGODB_URI, {
    useNewUrlParser: true,
    useUnifiedTopology: true,
    useFindAndModify: false,
    useCreateIndex: true,
  });
  console.log('database connected');
  connection.isConnected = db.connections[0].readyState;
}

export default dbConnect;
