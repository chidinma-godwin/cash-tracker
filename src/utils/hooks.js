import useSwr from 'swr';
import fetch from 'isomorphic-unfetch';

export const fetcher = url => fetch(url).then(response => response.json());

export const useUser = () => {
  const { data, error, mutate } = useSwr('/api/user', fetcher);
  const loading = !data && !error;
  const user = data?.user;
  return { user, loading, mutate };
};
