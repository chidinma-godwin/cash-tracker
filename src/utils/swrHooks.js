import useSwr from 'swr';

// CashTracker imports
import { userEndpoint, transactionEndpoint } from 'constants/endpoints';

export const fetcher = async url => {
  const res = await fetch(url);
  if (!res.ok) {
    const error = new Error();
    error.status = res.status;
    throw error;
  }
  return res.json();
};

export const useUser = () => {
  const { data, error, mutate } = useSwr(userEndpoint, fetcher);
  const loading = !data && !error;
  const user = data?.user;
  return { user, loading, mutate };
};

export const useTransaction = id => {
  const { data, error, mutate } = useSwr(
    `${transactionEndpoint}/${id}`,
    fetcher
  );
  const loading = !data && !error;
  const transactions = data?.transactions;
  return { transactions, loading, mutate };
};
