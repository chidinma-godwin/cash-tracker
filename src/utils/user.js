// CashTracker Imports
import User from 'models/User';

export async function createUser({ name, email, password, password2 }) {
  const user = await User.findOne({ email });
  if (user) {
    return new Error('A user with this email already exist');
  }
  if (password !== password2) {
    return new Error('Password do not match!');
  }
  const pendingInvitationsUsers = await User.find({ pendingRequests: email });
  const pendingInvitationsEmail =
    pendingInvitationsUsers &&
    pendingInvitationsUsers.map(pendingUser => pendingUser.email);
  const newUser = await User.create({
    username: name,
    email,
    password,
    ...(pendingInvitationsEmail && {
      pendingInvitations: pendingInvitationsEmail,
    }),
  });
  return { username: newUser.username, email: newUser.email };
}

export async function findUser(email, password) {
  const currentUser = await User.findOne({ email });
  if (currentUser && (await currentUser.passwordMatch(password))) {
    delete currentUser.password;
    return currentUser;
  }
  return new Error('Incorrect login credentials');
}
