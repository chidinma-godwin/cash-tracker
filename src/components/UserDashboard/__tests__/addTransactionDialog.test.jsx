// Cash Tracker imports
import userEvent from '@testing-library/user-event';
import { render, fireEvent, waitFor, screen } from 'testUtils';
import { useUser } from 'utils/hooks';
import AddTransactionDialog from '../addTransactionDialog';

jest.mock('utils/hooks', () => ({
  useUser: jest.fn(),
}));

describe('AddTransactionDialog component', () => {
  const requiredProps = {
    showDialog: true,
    handleCloseDialog: jest.fn(),
    handleTransactionAccount: jest.fn(),
    openSnackBar: false,
    handleCloseSnackBar: jest.fn(),
    errMsg: '',
  };
  beforeEach(() => {
    useUser.mockReturnValue({ user: { clientsEmail: ['test@email.com'] } });
  });
  it('matches snapshot when showDialog is true', () => {
    const { baseElement } = render(<AddTransactionDialog {...requiredProps} />);
    expect(baseElement).toMatchSnapshot();
  });
  it('matches snapshot when showDialog is false', () => {
    const { baseElement } = render(
      <AddTransactionDialog {...requiredProps} showDialog={false} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('calls handleTransactionAccount props when submit button is clicked', async () => {
    render(<AddTransactionDialog {...requiredProps} />);

    userEvent.type(screen.getByLabelText('Client Email'), 'test@email.com');
    userEvent.selectOptions(screen.getByLabelText('Currency'), 'NGN (₦)');

    const submitButton = screen.getByRole('button', { name: /Add/i });
    userEvent.click(submitButton);

    await waitFor(() => {
      expect(requiredProps.handleTransactionAccount).toHaveBeenCalled();
    });
  });
  it('calls handleCloseDialog props when cancel button is clicked', async () => {
    render(<AddTransactionDialog {...requiredProps} />);

    userEvent.type(screen.getByLabelText('Client Email'), 'test@email.com');
    userEvent.selectOptions(screen.getByLabelText('Currency'), 'NGN (₦)');

    const submitButton = screen.getByRole('button', { name: /Cancel/i });
    userEvent.click(submitButton);

    await waitFor(() => {
      expect(requiredProps.handleCloseDialog).toHaveBeenCalled();
    });
  });
  it('shows error alert if there is an error', () => {
    render(
      <AddTransactionDialog
        {...requiredProps}
        errMsg='Some error'
        openSnackBar
      />
    );

    expect(screen.getByRole('alert')).toHaveTextContent('Some error');
  });
  it('informs user that a request will be sent if the email is not in the options', async () => {
    render(<AddTransactionDialog {...requiredProps} />);
    const email = screen.getByLabelText('Client Email');

    userEvent.type(email, 'new@email.com');
    fireEvent.blur(email);

    await waitFor(() =>
      expect(
        screen.getByText(/There is no user with this email/i)
      ).toBeInTheDocument()
    );
  });
});
