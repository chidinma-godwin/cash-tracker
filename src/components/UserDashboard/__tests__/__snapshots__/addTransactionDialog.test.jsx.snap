// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTransactionDialog component matches snapshot when showDialog is false 1`] = `
<body
  style=""
>
  <div />
</body>
`;

exports[`AddTransactionDialog component matches snapshot when showDialog is true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="form-dialog-title"
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
          id="form-dialog-title"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            Add a new account
          </h2>
        </div>
        <form>
          <div
            class="MuiDialogContent-root"
          >
            
            <p
              class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Please enter the email of the user you want to start transactions with.
            </p>
            <div
              aria-expanded="false"
              class="MuiAutocomplete-root Mui-focused MuiAutocomplete-hasClearIcon"
              role="combobox"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-focused Mui-focused"
                  data-shrink="true"
                  for="clientEmail"
                  id="clientEmail-label"
                >
                  Client Email
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                >
                  <input
                    aria-autocomplete="list"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                    id="clientEmail"
                    spellcheck="false"
                    type="email"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Clear"
                      class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                      tabindex="-1"
                      title="Clear"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                    >
                      <span>
                        Client Email
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="true"
                for="currency"
                id="currency-label"
              >
                Currency
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <select
                  aria-invalid="false"
                  class="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  id="currency"
                >
                  <option
                    disabled=""
                    hidden=""
                    value=""
                  >
                    --Select Currency--
                  </option>
                  <option
                    value="AED"
                  >
                    AED (د.إ)
                  </option>
                  <option
                    value="AFN"
                  >
                    AFN (؋)
                  </option>
                  <option
                    value="ALL"
                  >
                    ALL (L)
                  </option>
                  <option
                    value="AMD"
                  >
                    AMD (֏)
                  </option>
                  <option
                    value="ANG"
                  >
                    ANG (ƒ)
                  </option>
                  <option
                    value="AOA"
                  >
                    AOA (Kz)
                  </option>
                  <option
                    value="ARS"
                  >
                    ARS ($)
                  </option>
                  <option
                    value="AUD"
                  >
                    AUD ($)
                  </option>
                  <option
                    value="AWG"
                  >
                    AWG (ƒ)
                  </option>
                  <option
                    value="AZN"
                  >
                    AZN (₼)
                  </option>
                  <option
                    value="BAM"
                  >
                    BAM (KM)
                  </option>
                  <option
                    value="BBD"
                  >
                    BBD ($)
                  </option>
                  <option
                    value="BDT"
                  >
                    BDT (৳)
                  </option>
                  <option
                    value="BGN"
                  >
                    BGN (лв)
                  </option>
                  <option
                    value="BHD"
                  >
                    BHD (.د.ب)
                  </option>
                  <option
                    value="BIF"
                  >
                    BIF (FBu)
                  </option>
                  <option
                    value="BMD"
                  >
                    BMD ($)
                  </option>
                  <option
                    value="BND"
                  >
                    BND ($)
                  </option>
                  <option
                    value="BOB"
                  >
                    BOB ($b)
                  </option>
                  <option
                    value="BOV"
                  >
                    BOV (BOV)
                  </option>
                  <option
                    value="BRL"
                  >
                    BRL (R$)
                  </option>
                  <option
                    value="BSD"
                  >
                    BSD ($)
                  </option>
                  <option
                    value="BTC"
                  >
                    BTC (₿)
                  </option>
                  <option
                    value="BTN"
                  >
                    BTN (Nu.)
                  </option>
                  <option
                    value="BWP"
                  >
                    BWP (P)
                  </option>
                  <option
                    value="BYN"
                  >
                    BYN (Br)
                  </option>
                  <option
                    value="BYR"
                  >
                    BYR (Br)
                  </option>
                  <option
                    value="BZD"
                  >
                    BZD (BZ$)
                  </option>
                  <option
                    value="CAD"
                  >
                    CAD ($)
                  </option>
                  <option
                    value="CDF"
                  >
                    CDF (FC)
                  </option>
                  <option
                    value="CHE"
                  >
                    CHE (CHE)
                  </option>
                  <option
                    value="CHF"
                  >
                    CHF (CHF)
                  </option>
                  <option
                    value="CHW"
                  >
                    CHW (CHW)
                  </option>
                  <option
                    value="CLF"
                  >
                    CLF (CLF)
                  </option>
                  <option
                    value="CLP"
                  >
                    CLP ($)
                  </option>
                  <option
                    value="CNY"
                  >
                    CNY (¥)
                  </option>
                  <option
                    value="COP"
                  >
                    COP ($)
                  </option>
                  <option
                    value="COU"
                  >
                    COU (COU)
                  </option>
                  <option
                    value="CRC"
                  >
                    CRC (₡)
                  </option>
                  <option
                    value="CUC"
                  >
                    CUC ($)
                  </option>
                  <option
                    value="CUP"
                  >
                    CUP (₱)
                  </option>
                  <option
                    value="CVE"
                  >
                    CVE ($)
                  </option>
                  <option
                    value="CZK"
                  >
                    CZK (Kč)
                  </option>
                  <option
                    value="DJF"
                  >
                    DJF (Fdj)
                  </option>
                  <option
                    value="DKK"
                  >
                    DKK (kr)
                  </option>
                  <option
                    value="DOP"
                  >
                    DOP (RD$)
                  </option>
                  <option
                    value="DZD"
                  >
                    DZD (دج)
                  </option>
                  <option
                    value="EEK"
                  >
                    EEK (kr)
                  </option>
                  <option
                    value="EGP"
                  >
                    EGP (£)
                  </option>
                  <option
                    value="ERN"
                  >
                    ERN (Nfk)
                  </option>
                  <option
                    value="ETB"
                  >
                    ETB (Br)
                  </option>
                  <option
                    value="ETH"
                  >
                    ETH (Ξ)
                  </option>
                  <option
                    value="EUR"
                  >
                    EUR (€)
                  </option>
                  <option
                    value="FJD"
                  >
                    FJD ($)
                  </option>
                  <option
                    value="FKP"
                  >
                    FKP (£)
                  </option>
                  <option
                    value="GBP"
                  >
                    GBP (£)
                  </option>
                  <option
                    value="GEL"
                  >
                    GEL (₾)
                  </option>
                  <option
                    value="GGP"
                  >
                    GGP (£)
                  </option>
                  <option
                    value="GHC"
                  >
                    GHC (₵)
                  </option>
                  <option
                    value="GHS"
                  >
                    GHS (GH₵)
                  </option>
                  <option
                    value="GIP"
                  >
                    GIP (£)
                  </option>
                  <option
                    value="GMD"
                  >
                    GMD (D)
                  </option>
                  <option
                    value="GNF"
                  >
                    GNF (FG)
                  </option>
                  <option
                    value="GTQ"
                  >
                    GTQ (Q)
                  </option>
                  <option
                    value="GYD"
                  >
                    GYD ($)
                  </option>
                  <option
                    value="HKD"
                  >
                    HKD ($)
                  </option>
                  <option
                    value="HNL"
                  >
                    HNL (L)
                  </option>
                  <option
                    value="HRK"
                  >
                    HRK (kn)
                  </option>
                  <option
                    value="HTG"
                  >
                    HTG (G)
                  </option>
                  <option
                    value="HUF"
                  >
                    HUF (Ft)
                  </option>
                  <option
                    value="IDR"
                  >
                    IDR (Rp)
                  </option>
                  <option
                    value="ILS"
                  >
                    ILS (₪)
                  </option>
                  <option
                    value="IMP"
                  >
                    IMP (£)
                  </option>
                  <option
                    value="INR"
                  >
                    INR (₹)
                  </option>
                  <option
                    value="IQD"
                  >
                    IQD (ع.د)
                  </option>
                  <option
                    value="IRR"
                  >
                    IRR (﷼)
                  </option>
                  <option
                    value="ISK"
                  >
                    ISK (kr)
                  </option>
                  <option
                    value="JEP"
                  >
                    JEP (£)
                  </option>
                  <option
                    value="JMD"
                  >
                    JMD (J$)
                  </option>
                  <option
                    value="JOD"
                  >
                    JOD (JD)
                  </option>
                  <option
                    value="JPY"
                  >
                    JPY (¥)
                  </option>
                  <option
                    value="KES"
                  >
                    KES (KSh)
                  </option>
                  <option
                    value="KGS"
                  >
                    KGS (лв)
                  </option>
                  <option
                    value="KHR"
                  >
                    KHR (៛)
                  </option>
                  <option
                    value="KMF"
                  >
                    KMF (CF)
                  </option>
                  <option
                    value="KPW"
                  >
                    KPW (₩)
                  </option>
                  <option
                    value="KRW"
                  >
                    KRW (₩)
                  </option>
                  <option
                    value="KWD"
                  >
                    KWD (KD)
                  </option>
                  <option
                    value="KYD"
                  >
                    KYD ($)
                  </option>
                  <option
                    value="KZT"
                  >
                    KZT (₸)
                  </option>
                  <option
                    value="LAK"
                  >
                    LAK (₭)
                  </option>
                  <option
                    value="LBP"
                  >
                    LBP (£)
                  </option>
                  <option
                    value="LKR"
                  >
                    LKR (₨)
                  </option>
                  <option
                    value="LRD"
                  >
                    LRD ($)
                  </option>
                  <option
                    value="LSL"
                  >
                    LSL (M)
                  </option>
                  <option
                    value="LTC"
                  >
                    LTC (Ł)
                  </option>
                  <option
                    value="LTL"
                  >
                    LTL (Lt)
                  </option>
                  <option
                    value="LVL"
                  >
                    LVL (Ls)
                  </option>
                  <option
                    value="LYD"
                  >
                    LYD (LD)
                  </option>
                  <option
                    value="MAD"
                  >
                    MAD (MAD)
                  </option>
                  <option
                    value="MDL"
                  >
                    MDL (lei)
                  </option>
                  <option
                    value="MGA"
                  >
                    MGA (Ar)
                  </option>
                  <option
                    value="MKD"
                  >
                    MKD (ден)
                  </option>
                  <option
                    value="MMK"
                  >
                    MMK (K)
                  </option>
                  <option
                    value="MNT"
                  >
                    MNT (₮)
                  </option>
                  <option
                    value="MOP"
                  >
                    MOP (MOP$)
                  </option>
                  <option
                    value="MRO"
                  >
                    MRO (UM)
                  </option>
                  <option
                    value="MRU"
                  >
                    MRU (UM)
                  </option>
                  <option
                    value="MUR"
                  >
                    MUR (₨)
                  </option>
                  <option
                    value="MVR"
                  >
                    MVR (Rf)
                  </option>
                  <option
                    value="MWK"
                  >
                    MWK (MK)
                  </option>
                  <option
                    value="MXN"
                  >
                    MXN ($)
                  </option>
                  <option
                    value="MXV"
                  >
                    MXV (MXV)
                  </option>
                  <option
                    value="MYR"
                  >
                    MYR (RM)
                  </option>
                  <option
                    value="MZN"
                  >
                    MZN (MT)
                  </option>
                  <option
                    value="NAD"
                  >
                    NAD ($)
                  </option>
                  <option
                    value="NGN"
                  >
                    NGN (₦)
                  </option>
                  <option
                    value="NIO"
                  >
                    NIO (C$)
                  </option>
                  <option
                    value="NOK"
                  >
                    NOK (kr)
                  </option>
                  <option
                    value="NPR"
                  >
                    NPR (₨)
                  </option>
                  <option
                    value="NZD"
                  >
                    NZD ($)
                  </option>
                  <option
                    value="OMR"
                  >
                    OMR (﷼)
                  </option>
                  <option
                    value="PAB"
                  >
                    PAB (B/.)
                  </option>
                  <option
                    value="PEN"
                  >
                    PEN (S/.)
                  </option>
                  <option
                    value="PGK"
                  >
                    PGK (K)
                  </option>
                  <option
                    value="PHP"
                  >
                    PHP (₱)
                  </option>
                  <option
                    value="PKR"
                  >
                    PKR (₨)
                  </option>
                  <option
                    value="PLN"
                  >
                    PLN (zł)
                  </option>
                  <option
                    value="PYG"
                  >
                    PYG (Gs)
                  </option>
                  <option
                    value="QAR"
                  >
                    QAR (﷼)
                  </option>
                  <option
                    value="RMB"
                  >
                    RMB (￥)
                  </option>
                  <option
                    value="RON"
                  >
                    RON (lei)
                  </option>
                  <option
                    value="RSD"
                  >
                    RSD (Дин.)
                  </option>
                  <option
                    value="RUB"
                  >
                    RUB (₽)
                  </option>
                  <option
                    value="RWF"
                  >
                    RWF (R₣)
                  </option>
                  <option
                    value="SAR"
                  >
                    SAR (﷼)
                  </option>
                  <option
                    value="SBD"
                  >
                    SBD ($)
                  </option>
                  <option
                    value="SCR"
                  >
                    SCR (₨)
                  </option>
                  <option
                    value="SDG"
                  >
                    SDG (ج.س.)
                  </option>
                  <option
                    value="SEK"
                  >
                    SEK (kr)
                  </option>
                  <option
                    value="SGD"
                  >
                    SGD (S$)
                  </option>
                  <option
                    value="SHP"
                  >
                    SHP (£)
                  </option>
                  <option
                    value="SLL"
                  >
                    SLL (Le)
                  </option>
                  <option
                    value="SOS"
                  >
                    SOS (S)
                  </option>
                  <option
                    value="SRD"
                  >
                    SRD ($)
                  </option>
                  <option
                    value="SSP"
                  >
                    SSP (£)
                  </option>
                  <option
                    value="STD"
                  >
                    STD (Db)
                  </option>
                  <option
                    value="STN"
                  >
                    STN (Db)
                  </option>
                  <option
                    value="SVC"
                  >
                    SVC ($)
                  </option>
                  <option
                    value="SYP"
                  >
                    SYP (£)
                  </option>
                  <option
                    value="SZL"
                  >
                    SZL (E)
                  </option>
                  <option
                    value="THB"
                  >
                    THB (฿)
                  </option>
                  <option
                    value="TJS"
                  >
                    TJS (SM)
                  </option>
                  <option
                    value="TMT"
                  >
                    TMT (T)
                  </option>
                  <option
                    value="TND"
                  >
                    TND (د.ت)
                  </option>
                  <option
                    value="TOP"
                  >
                    TOP (T$)
                  </option>
                  <option
                    value="TRL"
                  >
                    TRL (₤)
                  </option>
                  <option
                    value="TRY"
                  >
                    TRY (₺)
                  </option>
                  <option
                    value="TTD"
                  >
                    TTD (TT$)
                  </option>
                  <option
                    value="TVD"
                  >
                    TVD ($)
                  </option>
                  <option
                    value="TWD"
                  >
                    TWD (NT$)
                  </option>
                  <option
                    value="TZS"
                  >
                    TZS (TSh)
                  </option>
                  <option
                    value="UAH"
                  >
                    UAH (₴)
                  </option>
                  <option
                    value="UGX"
                  >
                    UGX (USh)
                  </option>
                  <option
                    value="USD"
                  >
                    USD ($)
                  </option>
                  <option
                    value="UYI"
                  >
                    UYI (UYI)
                  </option>
                  <option
                    value="UYU"
                  >
                    UYU ($U)
                  </option>
                  <option
                    value="UYW"
                  >
                    UYW (UYW)
                  </option>
                  <option
                    value="UZS"
                  >
                    UZS (лв)
                  </option>
                  <option
                    value="VEF"
                  >
                    VEF (Bs)
                  </option>
                  <option
                    value="VES"
                  >
                    VES (Bs.S)
                  </option>
                  <option
                    value="VND"
                  >
                    VND (₫)
                  </option>
                  <option
                    value="VUV"
                  >
                    VUV (VT)
                  </option>
                  <option
                    value="WST"
                  >
                    WST (WS$)
                  </option>
                  <option
                    value="XAF"
                  >
                    XAF (FCFA)
                  </option>
                  <option
                    value="XBT"
                  >
                    XBT (Ƀ)
                  </option>
                  <option
                    value="XCD"
                  >
                    XCD ($)
                  </option>
                  <option
                    value="XOF"
                  >
                    XOF (CFA)
                  </option>
                  <option
                    value="XPF"
                  >
                    XPF (₣)
                  </option>
                  <option
                    value="XSU"
                  >
                    XSU (Sucre)
                  </option>
                  <option
                    value="XUA"
                  >
                    XUA (XUA)
                  </option>
                  <option
                    value="YER"
                  >
                    YER (﷼)
                  </option>
                  <option
                    value="ZAR"
                  >
                    ZAR (R)
                  </option>
                  <option
                    value="ZMW"
                  >
                    ZMW (ZK)
                  </option>
                  <option
                    value="ZWD"
                  >
                    ZWD (Z$)
                  </option>
                  <option
                    value="ZWL"
                  >
                    ZWL ($)
                  </option>
                </select>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                  >
                    <span>
                      Currency
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Cancel
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Add
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
